pre_commit/languages/lua.py

```diff
@@ -0,0 +1,113 @@
+import contextlib
+import os
+import re
+from typing import Generator
+from typing import Sequence
+from typing import Tuple
+
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import PatchesT
+from pre_commit.envcontext import Var
+from pre_commit.hook import Hook
+from pre_commit.languages import helpers
+from pre_commit.prefix import Prefix
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import cmd_output
+
+ENVIRONMENT_DIR = 'lua_env'
+get_default_version = helpers.basic_get_default_version
+healthy = helpers.basic_healthy
+
+
+def install_environment(
+    prefix: Prefix,
+    version: str,
+    additional_dependencies: Sequence[str],
+) -> None:
+    helpers.assert_version_default('lua', version)
+    # luarocks, Lua's package manager, is separate from the runtime.
+    helpers.assert_version_default('luarocks', version)
+
+    envdir = _envdir(prefix, version)
+    with clean_path_on_failure(envdir):
+        with in_env(prefix, version):
+            _bootstrap_environment(envdir)
+
+            # Sometimes a rockspec is not available (like with a local hook).
```

Comment:
we should add a placeholder rockspec into the empty template -- for example see what I did for [dart](https://github.com/pre-commit/pre-commit/pull/2027/files#diff-8372cd96f090f7f0d3997aa2d99bf35c9d153aa89fa76d389343ca432251471b)